tap_s3_csv/transform.py

```diff
@@ -184,6 +183,33 @@ def transform_recur(self, data, schema, path, source_type=None):
                 Error(path, data, schema, logging_level=LOGGER.level))
             return False, None
 
+    def transform_schema_recur(self, schema):
+        if "type" not in schema:
+            return
+        types = schema["type"]
+        if not isinstance(types, list):
+            types = [types]
+            schema["type"] = types
+
+        # modify schema to put null as the last type to check for
+        # e.g. ['null', 'integer'] -> ['integer', 'null']
+        if "null" in types and types[-1] != "null":
+            types.remove("null")
+            types.append("null")
+
+        for typ in types:
+            if typ == "object" and "properties" in schema:
+                self._transform_schema_nested(schema.get("properties", {}))
```

Comment:
you wouldn't need a separate _transform_schema_nested if you loop here and call transform_schema_recur instead. would this work:
```
if typ == "object" and "properties" in schema:
    for property in schema['properties'].values():
        transform_schema_recur(property)
```